src/app/Library/CrudPanel/Traits/Validation.php

```diff
@@ -367,6 +250,77 @@ private function setupFieldValidation($field, $parent = false)
         }
     }
 
+    /**
+     * Return the messages for the fields and subfields in the current crud panel.
+     *
+     * @param  array  $fields
+     * @return array
+     */
+    private function getValidationMessagesFromFieldsAndSubfields($fields)
```

Comment:
Note to self: this hasn't changed at all, just moved in a different place in the same file.